bpf/k_tracer.c

```diff
@@ -674,17 +701,28 @@ int app_ingress(struct __sk_buff *skb) {
         return 0;
     }
 
-    unsigned char buf[12];
+    if (!tcp_syn(&tcp) || tcp_ack(&tcp)) {
+        return 0;
+    }
 
-    if (tcp_syn(&tcp) && !tcp_ack(&tcp)) {
-        bpf_skb_load_bytes(skb, tcp.hdr_len, &buf, 4);
+    if (skb->len - tcp.hdr_len < sizeof(tp_info_pid_t)) {
+        bpf_printk("SYN packet without tp info");
```

Comment:
these prints will have to be removed, but keeping them for now since this is disabled. bpf_dbg_printk doesnt' work because we need to fix it to not use bpf_get_pid_tid, but bpf_get_task info